idmapping-common/src/main/java/org/uniprot/api/idmapping/common/service/UniProtKBIdGroupByService.java

```diff
@@ -0,0 +1,86 @@
+package org.uniprot.api.idmapping.common.service;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.solr.client.solrj.response.FacetField;
+import org.uniprot.api.idmapping.common.model.IdMappingJob;
+import org.uniprot.api.idmapping.common.request.uniprotkb.UniProtKBIdMappingGroupByRequest;
+import org.uniprot.api.idmapping.common.response.model.IdMappingStringPair;
+import org.uniprot.api.rest.service.query.UniProtQueryProcessor;
+import org.uniprot.api.uniprotkb.common.service.groupby.GroupByService;
+import org.uniprot.api.uniprotkb.common.service.groupby.model.GroupByResult;
+import org.uniprot.api.uniprotkb.common.service.uniprotkb.UniProtEntryService;
+
+public abstract class UniProtKBIdGroupByService<T> {
+    private final GroupByService<T> groupByService;
+    private final UniProtEntryService uniProtEntryService;
+
+    protected UniProtKBIdGroupByService(
+            GroupByService<T> groupByService, UniProtEntryService uniProtEntryService) {
+        this.groupByService = groupByService;
+        this.uniProtEntryService = uniProtEntryService;
+    }
+
+    public GroupByResult getGroupByResult(
+            IdMappingJob idMappingJob, UniProtKBIdMappingGroupByRequest groupByRequest) {
```

Comment:
Add comment to statement and/or code block. Too complex to understand. Probably with example